scripts/utils/curriculum.py

```diff
@@ -112,3 +112,23 @@ def print_curriculum(self):
             print("Rest of the training probabilities:", self.rest_of_the_training_probabilities)
             print("Probabilities decay over time:", self.probabilities_decay_over_time)
         print("---------------")
+
+
+def curriculum_has_sp_types(curriculum:Curriculum) -> bool:
```

Comment:
I suggest changing this function to this:

```
def validate_curriculum_types(self, maybe_expected_types, excluded_types):
# Ensure at least one expected type is present in train_types
assert any(et in self.train_types for et in maybe_expected_types), \
    "Error: None of the expected types are present in train_types."

# Ensure no excluded types are present in train_types
assert not any(et in self.train_types for et in excluded_types), \
    "Error: One or more excluded types are present in train_types."
```

And then later in `train_helper.py` dont raise value just call:
```
curriculum.validate_curriculum_types(maybe_expected_types = [TeamType.SELF_PLAY_HIGH, ...]
```